download_data_set.py

```diff
@@ -0,0 +1,184 @@
+  "http://yann.lecun.com/exdb/mnist/train-images-idx3-ubyte.gz")
+  progress_bar("train_features.gz", train_features)
+  ungzip("train_features.gz", "train_features.ubytes")
+  
+  train_labels = requests.get(
+  "http://yann.lecun.com/exdb/mnist/train-labels-idx1-ubyte.gz")
+  progress_bar("train_labels.gz", train_labels)
+  ungzip("train_labels.gz", "train_labels.ubytes")
+
+  test_features = requests.get(
+  "http://yann.lecun.com/exdb/mnist/t10k-images-idx3-ubyte.gz")
+  progress_bar("test_features.gz", test_features)
+  ungzip("test_features.gz", "test_features.ubytes")
+  
+  test_labels = requests.get(
+  "http://yann.lecun.com/exdb/mnist/t10k-labels-idx1-ubyte.gz")
+  progress_bar("test_labels.gz", test_labels)
+  ungzip("test_labels.gz", "test_labels.ubytes")
+
+  print("Converting mnist ubytes images files into csv...")
+  print("This might take a while...")
+  print("Converting features images...")
+  convert("train_features.ubytes", "train_labels.ubytes",
+  "mnist_train.csv", 60000)
+  print("Converting label images...")
+  convert("test_features.ubytes", "test_labels.ubytes",
+  "mnist_test.csv", 10000)
+  clean()
+
+def electricity_consumption_dataset():
+  print("Download the electricty consumption example datasets")
+  electricity = requests.get("https://www.mlpack.org/datasets/examples/electricity-usage.csv")
+  progress_bar("electricity-usage.csv", electricity)
+
+def stock_exchange_dataset():
+  print("Download the stock exchange example datasets")
+  stock = requests.get("https://www.mlpack.org/datasets/examples/Google2016-2019.csv")
+  progress_bar("stock.csv", stock)
+
+def iris_dataset():
+  print("Downloading iris datasets...")
+  iris = requests.get("https://www.mlpack.org/datasets/iris.tar.gz")
+  progress_bar("iris.tar.gz", iris)
+  tar = tarfile.open("iris.tar.gz", "r:gz")
+  tar.extractall()
+  tar.close()
+  clean()
+  
+def all_datasets():
+  mnist_dataset()
+  electricity_consumption_dataset()
+  stock_exchange_dataset()
+  iris_dataset()
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser(
+        prog='Download dataset script',
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description=textwrap.dedent('''\
+        Dataset script!
+        --------------------------------
+        This script is used to download the dataset required to run
+        mlpack examples.
+        It can be used to download one dataset at the time,
+        or all of them at the same time.
+
+        Usage: --dataset_name dataset_name
+        Available options:
+        mnist : will download mnist dataset
+        electricity : will download electricty_consumption_dataset
+        stock : will download stock_exchange dataset
+        all : will download all datasets for all examples
+        '''))
+ 
+    parser.add_argument('--dataset_name', metavar="dataset name", type=str, help="Enter dataset name to download")
+    args = parser.parse_args()
+
+    if len(sys.argv)==1:
```

Comment:
```suggestion
    if len(sys.argv) == 1:
```
Sorry to be a bit pedantic. :)